### PR TITLE
fix objectId encoding/decoding on json objects

### PR DESCRIPTION
**Pull request by llfbandit**
Check that ObjectId convertion is applied to all objects that it should apply to #45

### CODE REVIEW

**Comment by karianna** [comment]
@llfbandit Pretty happy with this one as well although anything that touches ids make me nervous.  I'm wondering if further tests can be added around any edge cases here.

**Comment by llfbandit** [comment]
Yup, I understand, I've added only 1 test but is actually processed 3 times especially with one default behaviour and one with MongoClientWithObjectIdTest which add the property useObjectId.
Also there are already plenty of other tests with Id (with long values, random values, etc).
I'll try to dig but I think we're OK.

**Comment by llfbandit** [comment]
A further though about the use of useObjectId and its automatic behaviour is for nested objects, there's no recurrence made on objects (un-)wrap them.
I'm able to include it in my PR but I would ask for some validation before.

**Comment by johnoliver** [comment]
Again slight conflict, but happy to merge when fixed. Thanks again for the contribution

**Comment by karianna** [comment]
@llfbandit Couple of small conflicts left

File: vertx-mongo-client/src/main/asciidoc/groovy/index.adoc [hunk 1]
```diff
@@ -279,7 +279,7 @@ mongoClient.update("books", query, update, { res ->
 ----
 
 To specify if the update should upsert or update multiple documents, use `link:../../groovydoc/io/vertx/groovy/ext/mongo/MongoClient.html#updateWithOptions(java.lang.String,%20io.vertx.core.json.JsonObject,%20io.vertx.core.json.JsonObject,%20io.vertx.ext.mongo.UpdateOptions,%20io.vertx.core.Handler)[updateWithOptions]`
-and pass in an instance of `link:../dataobjects.html#UpdateOptions[UpdateOptions]`.
+and pass in an instance of `link:../../null/dataobjects.html#UpdateOptions[UpdateOptions]`.
```
Comment (karianna) [inline vertx-mongo-client/src/main/asciidoc/groovy/index.adoc]: Why ../null?

Comment (vietj) [inline vertx-mongo-client/src/main/asciidoc/groovy/index.adoc]: groovy doc generation bug ? how does it look like with other languages ?
